component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -0,0 +1,339 @@
+            AuthenticatedUser authenticatedUserObj =
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier((String) obj.
+                            get(Office365AuthenticatorConstants.ID));
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier(id);
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException | IOException e) {
+            throw new AuthenticationFailedException("Authentication process failed", e);
+        }
+    }
+
+    /**
+     * Get the oauth response
+     *
+     * @param oAuthClient   OAuthClient
+     * @param accessRequest Access request to get accessToken
+     * @return Response OAuthClientResponse.
+     * @throws AuthenticationFailedException
+     */
+    private OAuthClientResponse getOauthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws AuthenticationFailedException {
+        OAuthClientResponse oAuthResponse;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthSystemException | OAuthProblemException e) {
+            throw new AuthenticationFailedException("Exception while requesting access token " + e.getMessage(), e);
+        }
+        return oAuthResponse;
+    }
+
+    /**
+     * Get the access request
+     *
+     * @param tokenEndPoint Office365 authorize endpoint URL.
+     * @param clientId      The client ID of the client application that is registered in Azure AD.
+     * @param code          Authorization code to be used to obtain the access token.
+     * @param clientSecret  The value of the key that contains the client password.
+     * @param callbackUrl   Specifies the reply URL of the application.
+     * @return Response OAuthClientRequest.
+     * @throws AuthenticationFailedException
+     */
+    private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientId, String code, String clientSecret,
+                                                String callbackUrl) throws AuthenticationFailedException {
+        OAuthClientRequest accessRequest;
+        try {
+            accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint)
+                    .setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setCode(code)
+                    .setRedirectURI(callbackUrl)
+                    .setClientId(clientId)
+                    .setClientSecret(clientSecret)
+                    .buildBodyMessage();
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException("Exception while building request for request access token " +
+                    e.getMessage(), e);
+        }
+        return accessRequest;
+    }
+
+    /**
+     * Request user claims from user info endpoint.
+     *
+     * @param url         User info endpoint.
```

Comment:
Issue with Alignment

Reply:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-office365/pull/3